activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -822,6 +832,21 @@ def check_constraint(*args)
       def remove_check_constraint(*args)
         @base.remove_check_constraint(name, *args)
       end
+
+      private
+        def raise_on_if_exist_options(options)
+          unrecognized_option = options.keys.find do |key|
+            key == :if_exists || key == :if_not_exists
+          end
+          if unrecognized_option
+            conditional = unrecognized_option == :if_exists ? "if" : "unless"
+            message =
+              "Option #{unrecognized_option} will be ignored. If you are calling an expression like " +
+              "`t.column(.., #{unrecognized_option}: true)` from inside a change_table block, " +
+              "try a conditional clause instead, as in `t.column(..) #{conditional} t.column_exists?(..)`"
```

Comment:
I think it will be easier to read in the code if we use heredocs to write out the message like this:

```ruby
message = <<-MSG
  Option #{unrecognized_option} will be ignored. If you are calling an expression like
  `t.column(.., #{unrecognized_option}: true)` from inside a change_table block,
  try a conditional clause instead, as in `t.column(..) #{conditional} t.column_exists?(..)`
MSG

Reply:
and now all 3 lines are always the exact same length 👩🏼‍🍳💋